[addons] fix PVR disable/enable handle

AlwinEsch opened the pull request:
## Description

Fix PVR disable/enable handle. Before it used the Binary addon manager about but with the PVR was informed about change before them and not worked.

Also remove the no more used IsAddonEnabled from CBinaryAddonManager.

<!--- Provide a general summary of your change in the Pull Request title above -->
<!--- Describe your change in detail here -->

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your change -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc -->

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

ksooo reviewed (commented):
Not sure, but looking at the changes of this PR and at what I saw while I was debugging issues after switching profiles containing PVR addons, for me it looks like binary addon manager's data getting out of sync somehow. 

So, instead of letting binary addon manager call addon manager instead of using its own data (because they are broken), seems more like a work around than a real fix.

I suggest to fix binary addon manager to contain consistent data again.

AlwinEsch committed:
[addons] use for PVR disable/enable the normal addon manager

Also remove the no more used IsAddonEnabled from CBinaryAddonManager.

AlwinEsch committed:
[addons] remove no more needed/used functions from binary addon manager

On SettingConditions was one part used, but replaced to normal manager.
The rest was not needed and now removed.

AlwinEsch committed:
[addons] remove GetInstalledAddonInfo call where possible

This was on binary addon manager, this change remove it where easily possible
to change to normal addon manager.

On following changes the other parts becomes on bigger reworked ways
also changed (currently them need stil the BinaryAddonBasePtr where makes
harder).

AlwinEsch committed:
[addons] cleanup binary addon manager

This add docs about the two new functions and remove all no more needed
parts.

ksooo commented on xbmc/guilib/imagefactory.cpp:
```diff
@@ -37,9 +36,9 @@ IImage* ImageFactory::CreateLoader(const CURL& url)
 {
-  BinaryAddonBaseList addonInfos;
+  std::vector<AddonInfoPtr> addonInfos;
 
```
Remove this empty line? At other code sequences similar to this one there is no blank line.

ksooo commented on xbmc/peripherals/bus/virtual/PeripheralBusAddon.cpp:
```diff
@@ -410,9 +408,10 @@ void CPeripheralBusAddon::UpdateAddons(void)
+    auto GetAddon = [&addonId](const AddonInfoPtr& addon) { return addon->ID() == addonId; };
 
-    BinaryAddonBaseList::iterator it = std::find_if(newAddons.begin(), newAddons.end(), GetAddon);
+    std::vector<AddonInfoPtr>::iterator it =
```
`auto it`?

ksooo reviewed (commented):
Minors only. 

However, I cannot judge on the functional changes and there consequences. For example, removal of PVR addons from binary addon cache. But I trust you @AlwinEsch .

AlwinEsch committed:
[addons] check incompatible addons always by start

Before was them only done if "Auto update" was enabled in settings.
Now in case this is disabled it checks the system installed addons
and disable by incompatibility.

AlwinEsch committed:
[addons] use GetAddonInfo from normal manager where possible

This replace GetAddonInfo call to binary addon manager where possible.
On following changes it becomes also done on others, but here the smaller
changes about.

AlwinEsch committed:
[addons] rename getAddonInstance with a capital letter

Before was the first character lowercase where not match the code style.

AlwinEsch committed:
[addons] use CAddonInfo everywhere and prevent CBinaryAddonBase

Majorly is the handling changed, as CBinaryAddonBase becomes
created on background and register his now in binary addon manager.
On last destruct it becomes removed from there.

This thought to fix problems with to late or missing report on binary addon
manager and remove obsolete parts (them was mostly thought before with cpluff).

AlwinEsch reviewed (commented):
Thanks for the review, have updated this parts.
Also changed the function name to uppercase in last commit.
Further is a new patch added where fix the Kodi start if addon auto update is disabled.

ksooo reviewed (approved):
LGTM now.
